frame/contracts/src/wasm/runtime.rs

```diff
@@ -393,6 +393,50 @@ define_env!(Env, <E: Ext>,
 		}
 	},
 
+	// Transfer some value to another account.
+	//
+	// If the value transfer was succesful zero is returned. Otherwise a non-zero value is returned.
+	// The scratch buffer is not touched. The receiver can be a plain account or a contract.
+	//
+	// - account_ptr: a pointer to the address of the beneficiary account
+	//   Should be decodable as an `T::AccountId`. Traps otherwise.
+	// - account_len: length of the address buffer.
+	// - value_ptr: a pointer to the buffer with value, how much value to send.
+	//   Should be decodable as a `T::Balance`. Traps otherwise.
+	// - value_len: length of the value buffer.
+	ext_transfer(
+		ctx,
+		account_ptr: u32,
+		account_len: u32,
+		value_ptr: u32,
+		value_len: u32
+	) -> u32 => {
+		let callee: <<E as Ext>::T as frame_system::Trait>::AccountId =
+			read_sandbox_memory_as(ctx, account_ptr, account_len)?;
+		let value: BalanceOf<<E as Ext>::T> =
+			read_sandbox_memory_as(ctx, value_ptr, value_len)?;
+
+		let ext = &mut ctx.ext;
+		let result = ctx.gas_meter.with_nested(ctx.gas_meter.gas_left(), |nested_meter| {
+			match nested_meter {
+				Some(nested_meter) => {
+					ext.transfer(
+						&callee,
+						value,
+						nested_meter,
+					)
+				},
+				// there is not enough gas to allocate for the nested call.
+				None => Err(sp_runtime::DispatchError::Other("Out of gas.")),
+			}
+		});
```

Comment:
I think creating a nested gas meter passing whole available gas amount to it is essentially a no-op, isn't it? Can we use `ctx.gas_meter` without creating a nested meter?

Reply:
You are right. The nesting is not necessary as we are always using all the gas. I changed it.